### DISCUSSION TITLE
New Architecture Questions

### DISCUSSION THREAD

**Discussion by gtokman** [leotm/react-native-template-typescript#1531] [Q&A] [answered]
1. Do all the dependencies in the project support the new architecture? 

```
    "@react-native-async-storage/async-storage": "1.17.11",
    "@react-native-community/checkbox": "0.5.14",
    "@react-native-community/datetimepicker": "6.3.5",
    "@react-native-community/slider": "4.4.2",
    "@react-navigation/native": "6.1.6",
    "@react-navigation/native-stack": "6.9.12",
    "@tanstack/react-query": "4.24.10",
    "react": "18.2.0",
    "react-native": "0.71.1",
    "react-native-codegen": "0.71.3",
    "react-native-gesture-handler": "2.9.0",
    "react-native-reanimated": "3.0.0",
    "react-native-safe-area-context": "4.5.0",
    "react-native-screens": "3.19
```

The [docs](https://reactnative.dev/blog/2022/06/16/resources-migrating-your-react-native-library-to-the-new-architecture#migration-status-of-popular-libraries) say the current migrated libraries are the following: 

```
Migration Status of Popular Libraries
Libraries maintainers have been sharing with us [in the working group](https://github.com/reactwg/react-native-new-architecture/discussions/categories/libraries) the status of their migration effort, and we wanted to provide you with a quick overview:

[react-native-gesture-handler](https://github.com/reactwg/react-native-new-architecture/discussions/15): ✅ Migrated
[react-native-navigation](https://github.com/reactwg/react-native-new-architecture/discussions/17): 🏃‍♂️ Ongoing
[react-native-pager-view](https://github.com/reactwg/react-native-new-architecture/discussions/16): 🏃‍♂️ Ongoing
[react-native-reanimated](https://github.com/reactwg/react-native-new-architecture/discussions/14): ✅ Migrated. In the process of testing and profiling for performances
[react-native-screens](https://github.com/reactwg/react-native-new-architecture/discussions/13): 🏃‍♂️ Ongoing
[react-native-slider](https://github.com/reactwg/react-native-new-architecture/discussions/38): 🎬 Started
[react-native-template-new-architecture](https://github.com/reactwg/react-native-new-architecture/discussions/21): ✅ Migrated. Gradually adopting/testing more companion Libraries
[react-native-template-typescript](https://github.com/reactwg/react-native-new-architecture/discussions/22): ✅ Migrated
[react-native-webview](https://github.com/reactwg/react-native-new-architecture/discussions/19): 🎬 Started
```

2. Do you have to install pods using `RCT_NEW_ARCH_ENABLED=1 bundle exec pod install`?

**Comment by leotm** [comment, marked as answer]
hey @gtokman great question ^ 1.

- @react-native-async-storage/async-storage": "1.17.11", ✅ Migrated
- @react-native-community/checkbox": "0.5.14", ❌ Will likely remove
- @react-native-community/datetimepicker": "6.3.5", 🏃‍♂️ Ongoing (appears [migrated](https://github.com/react-native-datetimepicker/datetimepicker/commit/1d31948d1fe43f72d47c9e2e4537f1921f7f7b4a))
- @react-native-community/slider": "4.4.2", ✅ Migrated
- @react-navigation/native": "6.1.6", ✅ Migrated
- @react-navigation/native-stack": "6.9.12", ✅ Migrated
- @tanstack/react-query": "4.24.10", ✅ Migrated
- react": "18.2.0", ✅ Migrated
- react-native": "0.71.1", ✅ Migrated
- react-native-codegen": "0.71.3", ✅ Migrated
- react-native-gesture-handler": "2.9.0", ✅ Migrated
- react-native-reanimated": "3.0.0", ✅ Migrated
- react-native-safe-area-context": "4.5.0", ✅ Migrated
- react-native-screens": "3.19.0" ✅ Migrated

the blog's abit ood, i'll see if we can get it updated

in the meantime here's our current full list
- https://github.com/reactwg/react-native-new-architecture/discussions/6

---

2. in this template no need to add `RCT_NEW_ARCH_ENABLED=1` done for us here

https://github.com/leotm/react-native-template-new-architecture/blob/2cf45a15ec0a77104bd6bc4374f11b0c4257866a/ios/Podfile#L7

otherwise yes if starting from vanilla `npx react-native@latest init AwesomeProject` or `npx react-native@0.68.0 init AwesomeProject --version 0.68.0` we'd need to run with `RCT_NEW_ARCH_ENABLED=1` unless we add it to Podfile (further details [here](https://github.com/react-native-community/upgrade-support/issues/193#issuecomment-1125065823))
